README.md

```diff
@@ -1,37 +1,105 @@
-	)
-	if err != nil {
-		log.Fatal(err)
-	}
-	err = c.Gauge("request.duration", 1.2, nil, 1)
-	// ...
+// Create the client
+c, err := statsd.New("127.0.0.1:8125",
+    statsd.WithNamespace("flubber."),               // prefix every metric with the app name
+    statsd.WithTags([]string{"region:us-east-1a"}), // send the EC2 availability zone as a tag with every metric
+    // add more options here...
+)
+if err != nil {
+    log.Fatal(err)
 }
+
+// Send some metrics!
+err = c.Gauge("request.queue_depth", 12, nil, 1)
+err = c.Timing("request.duration", duration, nil, 1) // Uses a time.Duration!
+err = c.TimeInMilliseconds("request", 12, nil, 1)
+err = c.Incr("request.count_total", nil, 1)
+err = c.Decr("request.count_total", nil, 1)
+err = c.Count("request.count_total", 2, nil, 1)
 ```
 
-### Supported environment variables
+You can find a list of all the available options [here](https://godoc.org/github.com/DataDog/datadog-go/statsd#Option).
+
+## Supported environment variables
 
 - The client can use the `DD_AGENT_HOST` and (optionally) the `DD_DOGSTATSD_PORT` environment variables to build the target address if the `addr` parameter is empty.
-- If the `DD_ENTITY_ID` enviroment variable is found, its value will be injected as a global `dd.internal.entity_id` tag. This tag will be used by the Datadog Agent to insert container tags to the metrics. You should only `append` to the `c.Tags` slice to avoid overwriting this global tag.
+- If the `DD_ENTITY_ID` environment variable is found, its value will be injected as a global `dd.internal.entity_id` tag. This tag will be used by the Datadog Agent to insert container tags to the metrics. You should only `append` to the `c.Tags` slice to avoid overwriting this global tag.
+
+## Unix Domain Sockets Client
+
+The version 6 (and above) of the Agent accepts packets through a Unix Socket datagram connection.
+Details about the advantages of using UDS over UDP are available in our [docs](https://docs.datadoghq.com/developers/dogstatsd/unix_socket/).
+
+You can use this protocol by giving a `unix:///path/to/dsd.socket` address argument to the `New` constructor.
+
+
+### Blocking vs Asynchronous behavior
+
+When transporting DogStatsD datagram over UDS, two modes are available, "blocking" and "asynchronous".
+
+"blocking" allows for error checking but does not guarantee that calls to the library will return instantly. For example `client.Gauge(...)` might take 50ms or more to complete. If used in a hot path of your application, this behavior might significantly impact your performance.
+
+"asynchronous" does not allow for error checking but guarantees that calls are instantaneous (<1ms). This is similar to UDP behavior.
+
+Currently, in 2.x, "blocking" is the default behavior to ensure backward compatibility. To use the "asynchronous" behavior, use the `statsd.WithAsyncUDS()` option.
+**Version 3.0 will drop support for the blocking mode.** As such, we suggest enabling the "asynchronous" mode.
```

Comment:
Can we commit to this? Is this something that makes sense to disclose here and now?